metrotranscom/doorway
# feat: add google maps runtime config to public site

# Pull Request Template

## Issue Overview

This PR addresses #84 


## Description

This PR updates the public site to add a runtime config (see #81) and to update the Google Maps components to use use this config to resolve the Google Maps API key based on runtime values.  Other vars will follow, but this is the initial step taken to validate the approach.

## How Can This Be Tested/Reviewed?

Set the value of `GOOGLE_MAPS_API_KEY` to a dummy value in your `.env` file.  Build as normal with `yarn build`, then start with `yarn start`.  Open the site in your browser and verify that the map doesn't work.

Quit the running service, then modify your `.env` file to change `GOOGLE_MAPS_API_KEY` to an actual key.  Start the service again without rebuilding.  Open the site in your browser and verify that the map _does_ actually work.

Run `yarn test:unit` to validate unit tests.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
